src/main/java/org/openlmis/requisition/service/RequisitionTemplateService.java

```diff
@@ -37,11 +37,21 @@ public class RequisitionTemplateService {
    * Find template for the given program and facility type.
    */
   public RequisitionTemplate findTemplate(UUID programId, UUID facilityTypeId, boolean reportOnly) {
+
     RequisitionTemplate template = requisitionTemplateRepository
         .findTemplate(programId, facilityTypeId, reportOnly);
 
-    if (null == template) {
-      throw new ContentNotFoundMessageException(new Message(ERROR_REQUISITION_TEMPLATE_NOT_FOUND));
+    if (template == null) {
```

Comment:
please add additional condition "(template == null && reportOnly)

because in your code, if someone is trying to retrieve template with reportOnly false, it will be executed twice with the same parameters